### ISSUE TITLE
bootstrap tagging options

### ISSUE THREAD

**Issue by patrick-wright** [chef/chef-provisioning-aws#21]
this driver needs to the ability to tag ec2 instances.   fog adds a few default tags, perhaps we should do that too

**Comment by danieljimenez** [comment]
+1

**Comment by jtimberman** [comment]
We definitely should, this is essential functionality for our own internal project tracking.

**Comment by tyler-ball** [comment]
kitchen-ec2 handles this by having a `config[:tags]` element that is auto populated with `{:created_by => 'Test Kitchen'}` but can be expanded with user supplied tags.

I think we could do something similar.  @fnichol suggested having users supply this to the `with_driver` block because these tags will apply to more than just machines.  They apply to most AWS objects.

**Comment by danieljimenez** [comment]
+1, we use this for elastic search's EC2 discovery.

**Comment by lantrix** [comment]
I see there is now tagging being implemented in https://github.com/chef/chef-provisioning-aws/pull/190. I'd expect the next version of the gem would include this.

However, in the meantime I've just built `chef-provisioning-1.2.0.dev.0.gem` and the provisioning of EC2 instances successfully tagged them.

```
machine node['my_cookbook']['fqdn'] do
  machine_options({
    :aws_tags => {:MyKey => 'MyValue', :AnotherKey => 'AnotherValue'},
    :use_private_ip_for_ssh => true,
    :associate_public_ip_address => false,
    :source_key => node['my_cookbook']['aws_ssh_key_path'],
    :bootstrap_options => {
      :availability_zone => node['my_cookbook']['AZ'],
      :subnet_id => node['my_cookbook']['subnet'],
      :security_group_ids => [
        node['my_cookbook']['secgroups']['default']
      ],
      :private_ip_address => node['my_cookbook']['ip_address'],
      :key_name => node['my_cookbook']['aws_ssh_key_id'],
      :instance_type => node['my_cookbook']['instance_type']
    },
    :image_id => node['my_cookbook']['ami']
  })
end
```

**Comment by patrick-wright** [comment]
Glad it worked! I've added several specs to cover multiple cases. I merged #190 sometime last week, so it will be included in the next release.

**Comment by tyler-ball** [comment]
I'm looking to get a release out this week, but I keep putting it off because there are some conflicting dependencies in the ChefDK that I'll have to do the sacred _Conflicting Dependency Dance_ to solve

**Comment by tyler-ball** [comment]
This fix was released and is currently in 1.3.x branch of CPA
